CONTRIBUTING.md

```diff
@@ -92,28 +92,60 @@ It is possible to pass optional command line arguments to pytest, for example to
 run only certain tests using patterns (`-k`) or marker (`-m`).
 
 ```shell
-tox -e base -- <optional arguments>
+tox -e tests -- <optional arguments>
 ```
 
-Two important arguments are `--memcached-service` which allows to change the
-default of `localhost:11211` (memcached's default) to a different address, and
-`-n` which sets the number of parallel workers for pytest-xdist. There are two
-layers of parallelization in the tests. An inner one within the tests
-themselves, i.e. the parallelism in the algorithms, and an outer one by
-pytest-xdist. The latter is controlled by the `-n` argument. If you experience
-segmentation faults with the tests, try running them with `-n 0` to disable
-parallelization.
+There are a few important arguments:
 
-To test modules that rely on PyTorch, use:
+- `--memcached-service` allows to change the default of `localhost:11211` (memcached's default)
+  to a different address.
 
-```shell
-tox -e torch
-```
+  [Memcached](https://www.memcached.org/) is needed for testing
+  caching as well as speeding certain methods (e.g. Permutation Shapley).
+
+  To start memcached locally in the background with Docker use:
+
+  ```shell
+   run --name pydvl-memcache -p 11211:11211 -d memcached
+  ```
+
+- `-n` sets the number of parallel workers for 
+  [pytest-xdist](https://pytest-xdist.readthedocs.io/en/latest/).
+  
+  There are two layers of parallelization in the tests.
+  An inner one within the tests themselves, i.e. the parallelism in the algorithms,
+  and an outer one by pytest-xdist. The latter is controlled by the `-n` argument.
+  If you experience segmentation faults with the tests,
+  try running them with `-n 0` to disable parallelization.
+
+- `--slow-tests` enables running slow tests.
+
+### Markers
+
+We use a few different markers to differentiate between tests and runs
+groups of them of separately. Use `pytest --markers` to get a list and description
+of all available markers.
+
+Two important markers are:
+
+- `pytest.mark.slow` which is used to mark slow tests and skip them by default.
+
+  You can run additionaly run slow tests using: `pytest --slow-tests`
+
+- `pytest.mark.torch` which is used to mark tests that require PyTorch.
```

Comment:
Why do we treat pytorch separately? If that's because it's an optional dependency, so is ray, and so will be memcached. So maybe we need other marks for those? Or maybe we don't need this one for pytorch. If the module is installed, we run the tests, always, like ray, etc.

Reply:
I created an issue (#453) for this.